xzre.h

```diff
@@ -835,21 +842,23 @@ assert_offset(elf_handles_t, libc, 0x10);
 assert_offset(elf_handles_t, liblzma, 0x18);
 assert_offset(elf_handles_t, libcrypto, 0x20);
 
-typedef struct __attribute__((packed)) {
-	elf_handles_t *handles;
-	Elf64_Ehdr *libc_ehdr;
+typedef struct __attribute__((packed)) main_elf {
+	elf_handles_t *elf_handles;
+	Elf64_Ehdr *dynamic_linker_ehdr;
 	void **__libc_stack_end;
 } main_elf_t;
 
-assert_offset(main_elf_t, handles, 0x0);
-assert_offset(main_elf_t, libc_ehdr, 0x8);
+assert_offset(main_elf_t, elf_handles, 0x0);
+assert_offset(main_elf_t, dynamic_linker_ehdr, 0x8);
 assert_offset(main_elf_t, __libc_stack_end, 0x10);
 
+typedef struct backdoor_data backdoor_data_t;
```

Comment:
I ended up needing this forward declaration, and you can't make a forward typedef struct without giving the struct itself a name.

So I decided to follow the standard C library convention and give every struct a name, and add _t to all typdefs.